connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/ExecuteEventsManager.scala

```diff
@@ -192,13 +201,23 @@ case class ExecuteEventsManager(executeHolder: ExecuteHolder, clock: Clock) {
 
   /**
    * Post @link org.apache.spark.sql.connect.service.SparkListenerConnectOperationFinished.
+   * @param producedRowsCountOpt
+   *   Number of rows that are returned to the user. None is expected when the operation does not
```

Comment:
QQ: why not use `0` if not rows are returned?

Reply:
I think returning None would make it clear to us whether the corresponding query should have produced row or not, for example for a SELECT statement that has no result, we would put 0 here. And for query Like INSERT INTO, we would put a None instead of 0 here. This could better tell us whether a produced row is expected or not

Reply:
Agreed with [gjxdxh](https://github.com/gjxdxh) that it's clearer with None that no rows are expected for this operation.